### PR TITLE
[BugFix][Triton] fix rejection sampler when target_logits are NaN

### PR DESCRIPTION
**Pull request by SOMEONEUNSEEN**
### What this PR does / why we need it?

This PR fixes an issue in the rejection sampler where `target_logits` containing `NaN` or `Inf` values could cause errors or incorrect sampling. It cleans up `NaN` and `Inf` values in `target_logits` before applying logits processors.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

- vLLM version: v0.26.0
- vLLM main: https://github.com/vllm-project/vllm/commit/58d3918e3ea0a544ffedadad2ba84559e9c51d8f

### CODE REVIEW

**Comment by SOMEONEUNSEEN** [comment]
/cherry-pick releases/v0.26.0rc
[Bot]: cherry-pick completed successfully. New PR created: https://github.com/vllm-project/vllm-ascend/pull/14120

**Review by jianzs — CHANGES_REQUESTED** [review]
"target_logits contains NaN or INF" — isn't that a precision issue?

**Comment by SOMEONEUNSEEN** [comment]
> "target_logits contains NaN or INF" — isn't that a precision issue?

Thank you for your review. This PR is a preventive measure taken at the initial stage of sampling to intercept and prevent occasional NaNs from causing fatal infinite loop results. The root cause of the NaNs will be addressed in another PR for repair.

**Comment by jianzs** [comment]
> > "target_logits contains NaN or INF" — isn't that a precision issue?
> 
> Thank you for your review. This PR is a preventive measure taken at the initial stage of sampling to intercept and prevent occasional NaNs from causing fatal infinite loop results. The root cause of the NaNs will be addressed in another PR for repair.

It's better to tackling the root cause directly instead of introducing an ostrich-like workaround that hides the problem.

**Comment by SOMEONEUNSEEN** [comment]
> > > "target_logits contains NaN or INF" — isn't that a precision issue?
> > 
> > 
> > Thank you for your review. This PR is a preventive measure taken at the initial stage of sampling to intercept and prevent occasional NaNs from causing fatal infinite loop results. The root cause of the NaNs will be addressed in another PR for repair.
> 
> It's better to tackling the root cause directly instead of introducing an ostrich-like workaround that hides the problem.

You're right, this will mask other precision issues. I'll reconsider the fix on the main branch.
